st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -63,23 +67,52 @@ def init_vars(self, action_parameters):
             self.vars = self._get_rendered_vars(self.actionchain.vars,
                                                 action_parameters=action_parameters)
 
+    def validate(self):
+        """
+        Function which performs a simple compile time validation.
+
+        Keep in mind that some variables are only resolved during run time which means we can
+        perform only simple validation during compile / create time.
+        """
+        all_nodes = self._get_all_nodes(action_chain=self.actionchain)
+        on_success_nodes = self._get_all_on_success_nodes(action_chain=self.actionchain)
+        on_failure_nodes = self._get_all_on_failure_nodes(action_chain=self.actionchain)
+
+        # 1. Check all the "on-success" paths
+        for node_name in on_success_nodes:
+            valid_name = self._is_valid_node_name(all_node_names=all_nodes, node_name=node_name)
+
+            if not valid_name:
+                msg = 'Unable to find node with name "%s".' % (node_name)
```

Comment:
Getting the task name is possible, but getting the line number is a bit harder right now and would be more of a hack.